### DISCUSSION TITLE
Support for second order methods

### DISCUSSION THREAD

**Discussion by nemeshnorbert** [ceres-solver/ceres-solver#914] [Ideas]
Are there any plans to support second order methods? Currently we have only `ceres::FirstOrderFunction`. I would like to see interface where a user can supply not only the gradient but the hessian too.
Are there any mathematical grounds that could automate not only the evaluation of gradient but the hessian too? At the moment we can use Jets to automatically compute gradients.

**Comment by sandwichmaker** [comment]
There is no such plan currently. LBFGS is your best bet.
